src/app/components/audio-recordings/pages/analysis-results/analyses-results.component.ts

```diff
@@ -0,0 +1,185 @@
+  private rows = Array<AnalysisJobItemResult>();
+
+  public get analysisJob() {
+    // since this component is currently only used for default analysis jobs, we can hard code the attributes of this job
+    return new AnalysisJob({
+      name: "system",
+      id: "system",
+    });
+  }
+
+  public ngOnInit() {
+    this.getItems()
+      .then((items: AnalysisJobItemResult[]) => {
+        this.rows = items;
+        this.updateRows();
+      });
+  }
+
+  public updateRows = (): Observable<AnalysisJobItemResult[]> =>
+    this.rows$ = this.getRows();
+
+  public getRows() {
+    const data = new Observable<AnalysisJobItemResult[]>((observer) => {
+      observer.next(this.rows);
+    });
+    return data;
+  }
+
+  /**
+   * Fetches the `AnalysisJobItemResult` model from the baw-api
+   *
+   * @param model The id of the model that is requested from the baw-api
+   * @returns The complete `AnalysisJobItemResult` model of the requested item. If the item is not defined, the root path will be returned
+   */
+  public async getItems(model?: AnalysisJobItemResult): Promise<AnalysisJobItemResult[]> {
+    return new Promise((resolveTo) => {
+      const analysisJobId = this.analysisJob;
+      const response = lastValueFrom(this.api
+        .list(analysisJobId, this.audioRecording.id, model)
+        .pipe(first())
+        .pipe(
+          takeUntil(this.unsubscribe)
+        )
+      );
+
+      resolveTo(response);
+    });
+  }
+
+  // this method needs to be refactored
+  public toggleRow(model: AnalysisJobItemResult) {
+    this.getModelChildren(model.children ?? [], model).then((childItems) => {
+      this.addParentInformation(childItems, model);
+
+      if (model.open) {
+        // the relative path for all sub folders should include the folders name & two back slashes before and after the name
+        // this prevents false positives e.g. if the folder name is contained within another folders name
+        this.rows = this.rows.filter(item =>
+          // only the root item will not have a parent item, in this case, we can quickly determine that all folders need to be closed
+          !isInstantiated(item.parentItem) ||
+          !item.parentItem.path.includes(`/${model.name}/`) ||
+          item.name === model.name
+        );
+      } else {
+        this.rows.splice(this.rows.indexOf(model) + 1, 0, ...childItems);
+      }
+
+      model.open = !model.open;
+      this.updateRows();
+    });
+  }
+
+  // not all analysisJobItemResult models from the baw-api contain a path attribute therefore, we can not know the location of the item
+  // therefore, by adding information about the parent object, we can easily derive the relative path and hierarchy of a sub item
+  private addParentInformation(models: AnalysisJobItemResult[], parent: AnalysisJobItemResult) {
+    models.forEach(analysisJobItemResultItem => {
+      analysisJobItemResultItem.parentItem = parent;
+    });
+  }
```

Comment:
I'll be moving this to the `AnalysisJobItemResult` service

The reason is that the path attribute isn't emitted if a sub item is a item (not directory). I'm adding this information client side, but obviously, if this came from the api, this would not be needed
![image](https://user-images.githubusercontent.com/33742269/208003348-ee157a0d-2949-4bff-b8e7-c4a0f26c08a2.png)